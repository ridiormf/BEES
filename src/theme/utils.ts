type Space = number | string;

const utils = {
  marginHorizontal: (value: Space) => ({
    marginLeft: value,
    marginRight: value,
  }),
  marginVertical: (value: Space) => ({
    marginTop: value,
    marginBottom: value,
  }),
  paddingHorizontal: (value: Space) => ({
    paddingLeft: value,
    paddingRight: value,
  }),
  paddingVertical: (value: Space) => ({
    paddingTop: value,
    paddingBottom: value,
  }),
};

export default utils;
