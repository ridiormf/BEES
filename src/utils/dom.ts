export const getBody = () => {
  return document.getElementsByTagName('body')[0];
};
