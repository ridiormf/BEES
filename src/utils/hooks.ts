import React from 'react';
import { useLocation, useNavigate } from 'react-router-dom';

/**
 * Function that simulates the behavior of a class component constructor,
 * adapted to hooks, for functional components.
 *
 * @param {function} callback Callback which will be called as soon as the component is instantiated.
 */
export const useConstructor = <ValueType = void>(
  callback: () => ValueType,
): ValueType | void => {
  const componentWasBuilt = React.useRef<boolean>(false);
  const value = React.useRef<ValueType>();
  if (!componentWasBuilt.current) {
    componentWasBuilt.current = true;
    value.current = callback();
  }

  return value.current;
};

/**
 * Function that simulates the behavior of a class component method "componentDidMount",
 * adapted to hooks, for functional components.
 *
 * @param {function} callback Callback which will be called as soon as the component mount.
 */
export const useDidMount = (callback: () => void): void => {
  React.useEffect(callback, []);
};

/**
 * Function that simulates the behavior of a class component method "componentWillUnmount",
 * adapted to hooks, for functional components.
 *
 * @param {function} callback Callback which will be called as soon as the component will unmount.
 */
export const useWillUnmount = (callback: () => void): void => {
  React.useEffect(() => {
    return callback;
  }, []);
};

/**
 * Function that simulates the behavior of a class component method "componentDidUpdate",
 * adapted to hooks, for functional components.
 *
 * @param {function} callback Callback which will be called on each value change of variables declared as dependencies,
 * excluding the first render of the component.
 * @param {React.DependencyList} deps Dependency list to be listened.
 */
export const useDidUpdate = (
  callback: () => void,
  deps: React.DependencyList,
): void => {
  const firstRender = React.useRef<boolean>(true);
  React.useEffect(() => {
    if (firstRender.current) {
      firstRender.current = false;
    } else {
      callback();
    }
  }, deps);
};

/**
 * Function that use main hooks from "react-router-dom" to simplify usability.
 *
 * @returns {object} containing following attributes:
 * @type {function} "goBack": to return navigation one page.
 * @type {function} "go": to navigate for path.
 * @type {string} "pathname": current url pathname.
 */
export const useAppNavigation = () => {
  const navigate = useNavigate();
  const location = useLocation();

  const goBack = () => {
    navigate(-1);
  };

  const go = (path: string) => {
    navigate(path);
  };

  return {
    goBack,
    go,
    pathname: location.pathname,
  };
};
