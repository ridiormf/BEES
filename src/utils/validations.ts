/**
 * Function that checks if the full name is valid.
 * It will check:
 * 1. if the string has, at least, two names, and if all names has 2 letters or more;
 * 2. If the string not start or end with blank space;
 * 3. If it doesn't contain numbers; and
 * 4. If it doesn't contain two or more blank spaces between words
 *
 * @param {string} fullName String to be checked.
 * @returns {boolean} Return true if is a valid name, otherwise returns false.
 */
export const checkFullName = (fullName?: string): boolean => {
  if (!fullName) return false;

  const splittedNames = fullName.split(' ');

  return (
    splittedNames.length > 1 &&
    splittedNames.every((name) => name.length > 1 && !/\d/g.test(name))
  );
};
