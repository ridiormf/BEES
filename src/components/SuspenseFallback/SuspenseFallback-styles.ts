import { keyframes } from '@stitches/react';
import theme from '../../theme';

const Container = theme.styled('div', {
  alignSelf: 'stretch',
  flexGrow: 1,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  color: '$',
  variants: {
    color: {
      primary: {
        backgroundColor: '$primaryBackground',
      },
      secondary: {
        backgroundColor: '$secondaryBackground',
      },
    },
  },
});

const opacityAnimation = keyframes({
  '0%': { opacity: 0.1 },
  '50%': { opacity: 0.3 },
  '100%': { opacity: 0.1 },
});

const LoadingText = theme.styled('h1', {
  animation: `${opacityAnimation} 1000ms infinite`,
});

export const suspenseFallbackStyles = {
  styled: {
    Container,
    LoadingText,
  },
};
