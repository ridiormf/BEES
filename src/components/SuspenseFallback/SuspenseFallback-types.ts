import React from 'react';
import { suspenseFallbackStyles } from './SuspenseFallback-styles';

const {
  styled: { Container },
} = suspenseFallbackStyles;

export interface SuspenseFallbackProps
  extends React.ComponentProps<typeof Container> {}
