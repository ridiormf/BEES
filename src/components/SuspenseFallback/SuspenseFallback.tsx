import { suspenseFallbackStyles } from './SuspenseFallback-styles';
const {
  styled: { Container, LoadingText },
} = suspenseFallbackStyles;

const SuspenseFallback = () => {
  return (
    <Container>
      <LoadingText>Carregando...</LoadingText>
    </Container>
  );
};

export default SuspenseFallback;
