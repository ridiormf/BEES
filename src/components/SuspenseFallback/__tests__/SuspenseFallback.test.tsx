import { render } from '@testing-library/react';
import { Wrapper } from '../../../__tests__/utils';
import SuspenseFallback from '../SuspenseFallback';

describe('components ->', () => {
  describe('SuspenseFallback ->', () => {
    describe('snapshots ->', () => {
      it('Should match snapshot', () => {
        const { container } = render(<SuspenseFallback />, {
          wrapper: Wrapper,
        });
        expect(container).toMatchSnapshot();
      });
    });
  });
});
