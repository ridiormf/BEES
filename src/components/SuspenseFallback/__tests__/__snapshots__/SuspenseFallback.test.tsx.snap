// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components -> SuspenseFallback -> snapshots -> Should match snapshot 1`] = `
<div>
  <div
    class="c-dbppsk c-dbppsk-dnqaOS-color-primary"
  >
    <h1
      class="c-iYvCva"
    >
      Carregando...
    </h1>
  </div>
</div>
`;
